### PR TITLE
feat: Import `tenacity` library and use it in `cluster_read.py`

### PR DESCRIPTION
**Pull request by pablomh**
Use it to retry POST requests to Grafana which seems to close/reset connections under load.

[Tenacity](https://tenacity.readthedocs.io/en/latest/)

### CODE REVIEW

File: core/opl/cluster_read.py [hunk 1]
```diff
@@ -178,6 +179,7 @@ def _sanitize_target(self, target):
         target = target.replace("$Cloud", self.args.grafana_prefix)
         return target
 
+    @retry
```
Comment (jhutar) [inline core/opl/cluster_read.py]: How many times it will retry?

File: setup.py [hunk 2]
```diff
@@ -34,6 +34,7 @@
         "requests",
         "tabulate",
         "deepdiff",
+        "tenacity",
```
Comment (jhutar) [inline setup.py:37]: Pls also add it to `core/setup.py`

File: core/opl/cluster_read.py (hunk 1 above)
Comment (pablomh) [reply core/opl/cluster_read.py]: Is it better now?

**Review by jhutar — APPROVED** [review]

File: core/opl/cluster_read.py [hunk 3]
```diff
@@ -178,6 +179,8 @@ def _sanitize_target(self, target):
         target = target.replace("$Cloud", self.args.grafana_prefix)
         return target
 
+    # pylint: disable-next=undefined-variable
+    @retry(stop=(stop_after_delay(10) | stop_after_attempt(10)))  # noqa: F405
```
Comment (jhutar) [inline core/opl/cluster_read.py:183]: BTW never used `|` in python, thanks! https://stackoverflow.com/questions/5988665/pipe-character-in-python

Comment (pablomh) [reply core/opl/cluster_read.py:183]: :)

**Pull request merged**